docs/POST-MORTEM.md

```diff
@@ -0,0 +1,100 @@
+# Purpose
+
+This document will try to gather all the valuable information, debugging processes and insights developers could get
+during the process of Playground improvement.
+
+# [Major LSP Refactoring - PR #238](https://github.com/onflow/flow-playground/pull/238)
+
+## Problem
+
+The work encapsulated in that PR was targeted at the problem of state de-sync: contract could be updated/redeployed, but
+script importing said contract would not be aware of newly created fields or methods. As a result, user would need to
+refresh webpage to be able to use updated contract code.
+
+Said problem existed for quite some time in Playground and was not properly addressed because of lack of resources and
+priorities in other Developer Tools.
+
+## Hypothesis
+
+One of the components is caching or translates outdated code into LSP (_Language Server Protocol_).
+
+## Development Challenges
+
+- LSP Client and Server are nested deeply into React tree, which leads to unnecessary code updated and server/client
+  restarting on each state change.
+- Existing Webpack, eslint and Typescript compiling would create slow pipeline, resulting in 40 seconds update even on a
+  minuscule change
+- No `hot-reload`, meaning you will lose some changes you've done to the application state and slowing down testing
+  process
+- No automatic test pipeline
+
+## Approach
+
+### Make Typescript Less Strict
+
+Until the source of the problem is not identified we disabled Typescript strict checks to speed up linter (even for a
```

Comment:
minor writing suggestion:
Until the source of the problem is identified, we are we disabling Typescript strict checks to speed up linter..